engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -59,15 +60,43 @@ case class JobPreparationActor(executionData: WorkflowExecutionActorData,
   }
 
   def prepareJobExecutionActor(inputs: Map[LocallyQualifiedName, WdlValue]): Try[BackendJobPreparationSucceeded] = {
-    val jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, inputs)
-    Try(factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor)) map { BackendJobPreparationSucceeded(jobDescriptor, _) }
+    for {
+      evaluatedRuntimeAtts <- evaluateRuntimeAttributes(jobKey, expressionLanguageFunctions, inputs)
+      jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, evaluatedRuntimeAtts, inputs)
+    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))
   }
 
   // Split inputs map (= evaluated workflow declarations + coerced json inputs) into [init\.*].last
   private lazy val splitInputs = workflowDescriptor.backendDescriptor.inputs map { case (fqn, v) => splitFqn(fqn) -> v }
 
-  def resolveAndEvaluate(jobKey: BackendJobDescriptorKey,
-                         wdlFunctions: WdlStandardLibraryFunctions): Try[Map[LocallyQualifiedName, WdlValue]] = {
+  def evaluateRuntimeAttributes(jobKey: BackendJobDescriptorKey,
+                                wdlFunctions: WdlStandardLibraryFunctions,
+                                evaluatedInputs: Map[LocallyQualifiedName, WdlValue]): Try[Map[String, WdlValue]] = {
+    val tryInputs = evaluatedInputs.mapValues(Try(_))
+    val mapOfTries = jobKey.call.task.runtimeAttributes.attrs mapValues {
+      expr => expr.evaluate(buildMapBasedLookup(tryInputs), wdlFunctions)
+    }
+    TryUtil.sequenceMap(mapOfTries).map(addDefaultsToAttributes(_))
+  }
+
+  def addDefaultsToAttributes(specifiedAttributes: Map[LocallyQualifiedName, WdlValue]): Map[LocallyQualifiedName, WdlValue] = {
+    import WdlValueJsonFormatter._
+    import spray.json._
+
+    val workflowOptions = executionData.workflowDescriptor.backendDescriptor.workflowOptions
+    def isUnspecifiedAttribute(name: String) = !specifiedAttributes.contains(name)
+
+    val missing = factory.runtimeAttributeDefinitions.filter(x => isUnspecifiedAttribute(x.name))
+    val defaults = missing map { x => (x, workflowOptions.getDefaultRuntimeOption(x.name)) } collect {
+      case (runtimeAttributeDefinition, Success(jsValue)) => runtimeAttributeDefinition.name -> jsValue.convertTo[WdlValue]
```

Comment:
The `convertTo` is going to make a `WdlValue` out of the JSON data, but what happens if it's not of an appropriate type for the attribute?